src/SIL.XForge.Scripture/Services/SFProjectService.cs

```diff
@@ -653,40 +691,35 @@ public async Task<string> CheckLinkSharingAsync(string curUserId, string shareKe
                     project.CheckingConfig.CheckingEnabled && project.CheckingConfig.ShareEnabled
                 },
                 { SFProjectRole.SFObserver, project.TranslateConfig.ShareEnabled },
-                { SFProjectRole.Reviewer, project.TranslateConfig.ShareEnabled },
+                { SFProjectRole.Reviewer, project.TranslateConfig.ShareEnabled }
             }
                 .Where(entry => entry.Value)
                 .Select(entry => entry.Key)
                 .ToArray();
 
-            if (availableRoles.Contains(projectSecretShareKey.ProjectRole))
+            if (!availableRoles.Contains(projectSecretShareKey.ProjectRole))
             {
-                await AddUserToProjectAsync(
-                    conn,
-                    projectDoc,
-                    userDoc,
-                    projectSecretShareKey.ProjectRole,
-                    projectSecretShareKey.Key
-                );
-                return projectId;
+                return false;
             }
         }
-        // Look for a valid specific user share key.
-        if (
-            projectSecretShareKey.ExpirationTime > DateTime.UtcNow
+        else if (
+            projectSecretShareKey.ExpirationTime < DateTime.UtcNow
             && projectSecretShareKey.ShareLinkType == ShareLinkType.Recipient
         )
         {
-            await AddUserToProjectAsync(
-                conn,
-                projectDoc,
-                userDoc,
-                projectSecretShareKey.ProjectRole,
-                projectSecretShareKey.Key
-            );
-            return projectId;
+            return false;
         }
-        throw new ForbiddenException();
+        return true;
+    }
+
+    public SFProjectSecret GetProjectSecret(string shareKey)
```

Comment:
I think it would make it clearer to rename this method to `GetProjectSecretByShareKey`, as usually when there is a single string parameter it is assumed to be the id.